src/Avalonia.Controls/SizeChangedEventArgs.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Avalonia.Interactivity;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// Provides data specific to a SizeChanged event.
+    /// </summary>
+    public class SizeChangedEventArgs : RoutedEventArgs
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SizeChangedEventArgs"/> class.
+        /// </summary>
+        /// <param name="routedEvent">The routed event associated with these event args.</param>
+        public SizeChangedEventArgs(RoutedEvent? routedEvent)
+            : base (routedEvent)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SizeChangedEventArgs"/> class.
+        /// </summary>
+        /// <param name="routedEvent">The routed event associated with these event args.</param>
+        /// <param name="source">The source object that raised the routed event.</param>
+        public SizeChangedEventArgs(RoutedEvent? routedEvent, IInteractive? source)
+            : base(routedEvent, source)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SizeChangedEventArgs"/> class.
+        /// </summary>
+        /// <param name="routedEvent">The routed event associated with these event args.</param>
+        /// <param name="source">The source object that raised the routed event.</param>
+        /// <param name="previousSize">The previous size (or bounds) of the object.</param>
+        /// <param name="newSize">The new size (or bounds) of the object.</param>
+        public SizeChangedEventArgs(
+            RoutedEvent? routedEvent,
+            IInteractive? source,
+            Size previousSize,
+            Size newSize)
+            : base(routedEvent, source)
+        {
+            PreviousSize = previousSize;
+            NewSize = newSize;
+            HeightChanged = newSize.Height != previousSize.Height;
+            WidthChanged = newSize.Width != previousSize.Width;
```

Comment:
WPF Source has an interesting comment here:

>/// and flags indicating whether Width or Height actually changed. <para/>
>/// These flags are provided to avoid common mistake of comparing new and old values
>/// since simple compare of double-precision numbers can yield "not equal" when size in fact
>/// didn't change (round off errors accumulated during input processing and layout calculations
>/// may result in very small differencies that are considered "the same layout").

https://github.com/dotnet/wpf/blob/bba0c14e636cb6fa3e6108e2d7a81b70cca5c139/src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/FrameworkElement.cs#L3324-L3332

@maxkatz6 Do you know the standard way in Avalonia of checking "approximately equal" for layout purposes? I can probably dig up the code but if you know it already that would be faster.

Reply:
At least this const can be used https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Layout/LayoutHelper.cs#L16

Reply:
Thanks, will update later today.

Reply:
Updated. Everything should be all set now.